Library/Homebrew/utils/analytics.rb

```diff
@@ -313,8 +313,12 @@ def default_tags_influx
       sig { returns(T::Hash[Symbol, String]) }
       def default_fields_influx
         @default_fields_influx ||= begin
-          version = HOMEBREW_VERSION.match(/^[\d.]+/)[0]
-          version = "#{version}-dev" if HOMEBREW_VERSION.include?("-")
+          version = if HOMEBREW_VERSION.start_with?(">=")
+            "unknown"
+          else
+            suffix = "-dev" if HOMEBREW_VERSION.include?("-")
+            HOMEBREW_VERSION.match(/^[\d.]+/)[0] + suffix.to_s
+          end
```

Comment:
Would it be nicer to just fall through to `unknown` whenever the regex fails, instead of making an assumption about the form of the non-regex case?

That will prevent this from breaking in case we change the form of the unknown version in the future.

Reply:
Agreed.